decidim-core/app/controllers/concerns/decidim/direct_upload.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+module Decidim
+  module DirectUpload
+    extend ActiveSupport::Concern
+
+    included do
+      include Decidim::NeedsOrganization
+      skip_before_action :verify_organization
+
+      before_action :check_organization!,
+                    :check_authenticated!,
+                    :check_user_belongs_to_organization,
+                    :validate_direct_upload
+    end
+
+    protected
+
+    def verify_organization; end
+
+    def validate_direct_upload
+      return if current_admin.present?
```

Comment:
Just a side note that first I was wondering why there are these checks for `current_admin` but then I realized it is for the system admin panel.

Just adding a note that this was unclear to me just by browsing through the code. If you see it reasonable, I would maybe suggest extracting these checks to a separate method, e.g. `system_admin?` which would have made it clear for me.

Reply:
Added a comment: `# We skip the validation if we are in system panel. `current_admin` refers to the main system admin user.`